wrappers/javascript/test/basic_test.ts

```diff
@@ -21,6 +21,50 @@ describe('Automerge', () => {
             assert.deepEqual(doc2, {  hello: "world", big: "little", zip: "zop", app: "dap" })
         })
 
+        it('can detect an automerge doc with isAutomerge()', () => {
+            let doc1 = Automerge.from({ sub: { object: true } })
+            assert(Automerge.isAutomerge(doc1))
+            assert(!Automerge.isAutomerge(doc1.sub))
+            assert(!Automerge.isAutomerge("String"))
+            assert(!Automerge.isAutomerge({ sub: { object: true }}))
+            assert(!Automerge.isAutomerge(undefined))
+        })
+
+        it('it should recursively freeze the document if requested', () => {
+            let doc1 = Automerge.init({ freeze: true } )
```

Comment:
This feels like an API that will be surprising to javascript programmers, is it possible to hook into `Object.freeze` at all so folks can just do:

```
let doc1 = Automerge.init()
Object.freeze(doc1)
```

Reply:
Ok.  Fixed.